src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp

```diff
@@ -0,0 +1,140 @@
+ * +D^{(\hat{2})}\left(x^3\frac{\partial x^2}{\partial\xi^{\hat{1}}}\right)
+ * -D^{(\hat{2})}\left(x^2\frac{\partial x^3}{\partial\xi^{\hat{1}}}\right)\\
+ * 2J\frac{\partial \xi^{\hat{1}}}{\partial x^2}&=
+ * D^{(\hat{2})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{2})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{3})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{2}}}{\partial x^2}&=
+ * D^{(\hat{1})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{1})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{3})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{1}}}\right)\\
+ * 2J\frac{\partial \xi^{\hat{3}}}{\partial x^2}&=
+ * D^{(\hat{1})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{1})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{2}}}\right)
+ * +D^{(\hat{2})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{2})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{1}}}{\partial x^3}&=
+ * D^{(\hat{2})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{2})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{3})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{2}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{2}}}{\partial x^3}&=
+ * D^{(\hat{1})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{1})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{3})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{3}}}{\partial x^3}&=
+ * D^{(\hat{1})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{1})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * +D^{(\hat{2})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{2})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{1}}}\right)
+ *
+ * \f}
+ *
+ * This ensures that:
+ *
+ * \f{align}{
+ *
+ * \partial_{\hat{\imath}}\left(J\frac{\partial\xi^{\hat{\imath}}}
+ * {\partial x^i}\right)=0.
+ *
+ * \f}
+ */
+template <size_t Dim>
+void metric_identity_jacobian(
+    gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        det_jac_times_inverse_jacobian,
+    const Mesh<Dim>& mesh,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords,
+    const Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        inverse_jacobian) noexcept;
+}  // namespace dg
```

Comment:
For my understanding, why is this in the `dg` namespace? Isn't this just a general concept relating to mappings, and unrelated to DG? I don't object to the location, I mostly want to understand your reasoning. We can easily move it outside the namespace in the future if we want to use this function for non-DG code.

Reply:
The internals basically assume DG because of the differentiation matrices. There are other ways of achieving free-streaming preservation. In FD codes it's usually done by subtracting off the divergence of J d xi^{hat{i}}/dx^i instead of trying to modify the calculation of the term. Doing it by subtraction is a lot easier when doing adaptive-order FD, I think. Do you think this would be worth adding to the doxygen comment? I sort of like the idea of keeping history of "why does this exist this way?" recorded, and while the GitHub comments and/or commit history sort of work, it's a lot more difficult to find than a comment in the code.

Reply:
To me the use of the differentiation matrices implies "spectral" but not necessarily "DG". But as this is pretty pedantic I think we should leave the namespace as-is. I do agree that adding a paragraph to the dox explaining what you've just told me above would be quite useful 👍